discovery-infra/test_infra/factory.py

```diff
@@ -0,0 +1,46 @@
+from netaddr import IPNetwork
+
+from test_infra import utils, consts
+from test_infra.assisted_service_api import InventoryClient
+from test_infra.controllers.nat_controller import NatController
+from test_infra.controllers.node_controllers import TerraformController
+from test_infra.controllers.proxy_controller.proxy_controller import ProxyController
+from test_infra.helper_classes.cluster import Cluster
+from test_infra.helper_classes.nodes import Nodes
+from test_infra.tools.assets import LibvirtNetworkAssets
+from tests.config import TerraformConfig, ClusterConfig
+
+
+def nodes_factory(config: TerraformConfig, nat: NatController):
```

Comment:
I remember us having a similar code, no?
Or maybe it just creates the controller and not Nodes